### DISCUSSION TITLE
Namespacing

### DISCUSSION THREAD

**Discussion by skurvish** [trobfab/fabrik#83] [General]
At some point all of Fabrik will need to be properly namespaced. Some of it is partially namespaced with shortcuts implemented by Hugh I presume. This will work at present with Joomla 4. Joomla 5 however is far more strict.

For example, in the form fields.xml files there is an addfieldpath parameter which points to the fields in the admin models directory. This parameter will no longer be supported in Joomla 5 and is replaced with addfieldprefix which defines a namespace for where to find the fields.

This will be a significant undertaking. When to do it??

**Comment by trob** [comment]
At the moment I would stay to "Just make it work for J!4".

Maybe with some sort of (semi)stable Fabrik4 it will make sense to have an additional Fabrik5 (or Fabrik4-without-deprecated-stuff) branch where things can be done or at least documented while running into them.
E.g. Factory::getDbo(); is deprecated, too
I don't know how far Joomla's J!5 definitions are fix at all.

**Comment by joomlahenk** [comment]
Indeed: Just make it work for J!4 and don't spend type to get it for J!5, so you have untill July 2027

**Comment by skurvish** [comment]
I agree, I just wanted to make sure this was noted. Perhaps by 2027 Sophist's new replacement will be ready?

**Comment by trob** [comment]
I'm going to be retired, so not sooo interested in 2027

**Comment by skurvish** [comment]
Happened to notice that a lot of work was done on Fabrik to namespace it for J4 back in 2019. I suspect at some point when the J! team changed tracks on the J4 development and MAT killed any further J4 development that these changes were never integrated.

Might be worth having a look to see if the namespace work was complete and whether it could be merged into this branch.

**Comment by joomlahenk** [comment]
Yes, we can take a look at it but not more than that. We have our fabrik4 already updated for that. Sure we will find some optimizations, but there are more important updates for fabrik4 still to do.  We also need to do much more testing.

Merging this in our branch can give more trouble than expected.

 

Van: skurvish ***@***.***> 
Verzonden: Tuesday, 9 August 2022 14:50
Aan: joomlahenk/fabrik ***@***.***>
CC: joomlahenk ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [joomlahenk/fabrik] Namespacing (Discussion #83)

 

Happened to notice that a lot of work was done on Fabrik to namespace it for J4 back in 2019. I suspect at some point when the J! team changed tracks on the J4 development and MAT killed any further J4 development that these changes were never integrated.

Might be worth having a look to see if the namespace work was complete and whether it could be merged into this branch.
